ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OptimizeQueueService.java

```diff
@@ -484,117 +484,137 @@ public String queueName() {
-                  LOG.info("this plan started {}, {}", attemptId, jobId);
-                  while (retry <= retryTime) {
-                    LOG.debug("start get plan task retry {}", retry);
-                    retry++;
-                    long planStartTime = System.currentTimeMillis();
-                    List<OptimizeTaskItem> tasks = plan(planStartTime);
-                    if (CollectionUtils.isNotEmpty(tasks)) {
-                      isHaveTask = true;
-                      break;
-                    }
-
-                    try {
-                      Thread.sleep(retryInterval);
-                    } catch (InterruptedException e) {
-                      LOG.error("Internal Thread Interrupted", e);
-                    }
-                  }
+            LOG.info("this plan started {}, {}", attemptId, jobId);
+            while (retry <= retryTime) {
+              LOG.debug("start get plan task retry {}", retry);
+              retry++;
+              long planStartTime = System.currentTimeMillis();
+              List<OptimizeTaskItem> tasks = plan(planStartTime);
+              if (CollectionUtils.isNotEmpty(tasks)) {
+                isHaveTask = true;
+                break;
+              }
 
-                  // no task have planned
-                  if (!isHaveTask) {
-                    LOG.debug("The queue {} has retry {} times, no task have planned",
-                        optimizeQueue.getOptimizeQueueMeta().getQueueId(),
-                        retryTime);
-                  }
-                } catch (Throwable t) {
-                  LOG.error("failed to plan", t);
-                  throw t;
-                } finally {
-                  LOG.info("this plan end {}, cost {} ms, retry {}",
-                      attemptId, System.currentTimeMillis() - threadStartTime, retry);
-                  if (planThreadStarted.compareAndSet(true, false)) {
-                    lock();
-                    try {
-                      planThreadCondition.signalAll();
-                    } finally {
-                      unlock();
-                    }
-                  }
-                }
-              });
-              planThread.setName(
-                  "Optimize Plan Thread Queue-" + optimizeQueue.getOptimizeQueueMeta().getQueueId());
-              planThread.start();
-            } else {
-              lock();
               try {
-                // if timeout, return null
-                if (!planThreadCondition.await(waitTime - (System.currentTimeMillis() - startTime),
-                    TimeUnit.MILLISECONDS)) {
-                  LOG.debug("The queue {} has no task have planned", optimizeQueue.getOptimizeQueueMeta().getQueueId());
-                  return null;
-                }
-              } finally {
-                unlock();
+                Thread.sleep(retryInterval);
+              } catch (InterruptedException e) {
+                LOG.error("Internal Thread Interrupted", e);
```

Comment:
We may need to break from the loop when got an interrupted exception.

Reply:
Yes, it should be breaked.